org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/messages.properties

```diff
@@ -594,14 +594,14 @@
 585 = Type safety: Elements of type {0} need unchecked conversion to conform to {1}
 
 ### SOURCE LEVEL
-590 = Syntax error, type parameters are only available if source level is 1.5 or greater
-591 = Syntax error, static imports are only available if source level is 1.5 or greater
-592 = Syntax error, 'for each' statements are only available if source level is 1.5 or greater
-593 = Syntax error, parameterized types are only available if source level is 1.5 or greater
-594 = Syntax error, enum declarations are only available if source level is 1.5 or greater
-595 = Syntax error, varargs are only available if source level is 1.5 or greater
-596 = Syntax error, annotations are only available if source level is 1.5 or greater
-597 = Syntax error, annotation declarations are only available if source level is 1.5 or greater
+590 = Syntax error, type parameters are only available if source level is 1.8 or greater
```

Comment:
These error messages will never be emitted hereafter - right ? So does it call for an edit at all ?

Reply:
I was looking for all `1.x` strings without checking whether there is code that uses them or not. The amount of work required by such checks would otherwise exceed my available time.

Reply:
Removing problem messages without removing the corresponding IProblem constants creates an inconsistency. To avoid that inconsistency I second @srikanth-sankaran that removing just the messages is not a sure win.

Reply:
> Removing problem messages without removing the corresponding IProblem constants creates an inconsistency

I didn't realized that it was a call to remove constants. I will check if this is possible (API?) and remove them if it will be OK.

Reply:
> I didn't realized that it was a call to remove constants. I will check if this is possible (API?) and remove them if it will be OK.

It is API, and JDT/UI is our next door client referencing many of these constants. That's why I'd go with @srikanth-sankaran : don't bother right now, keep constants along with their messages.

BTW, I've been dreaming about a shortcut to navigate from IProblem to messages.properties and back ... this would come in handy when it's time to remove :)

Reply:
OK, conclusion here: I will restore messages. Whether IProblem constants (API!) could be removed/deprecated etc could be decided later.